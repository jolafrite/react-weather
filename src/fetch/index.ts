export * from './load-weather/load-weather.fetch';
export * from './search-weather/search-weather.fetch';
