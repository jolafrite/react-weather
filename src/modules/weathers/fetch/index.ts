export * from './load-5-days-weather-forecast/load-5-days-weather-forecast.fetch';
export * from './load-weather/load-weather.fetch';
export * from './search-weather/search-weather.fetch';
