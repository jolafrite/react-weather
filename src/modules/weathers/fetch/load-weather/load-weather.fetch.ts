import Axios from "axios";
import {
  IWeatherWind,
  IWeatherClouds,
  IWeatherMain,
  IWeatherWeather,
  IWeatherRain,
  IWeatherSnow
} from "../../models";

export interface ILoadWeatherFetchResponse {
  coord: {
    lon: number;
    lat: number;
  };
  weather: IWeatherWeather[];
  base: "stations";
  main: IWeatherMain;
  visibility: number;
  wind: IWeatherWind;
  clouds: IWeatherClouds;
  dt: number;
  sys: {
    type: number;
    id: number;
    message: number;
    country: string;
    sunrise: number;
    sunset: number;
  };
  rain?: IWeatherRain | null;
  snow?: IWeatherSnow | null;
  timezone: number;
  id: number;
  name: string;
  cod: number;
}

export const loadWeatherFetch = async (id: number) => {
  const { data } = await Axios.request<ILoadWeatherFetchResponse>({
    method: "get",
    url: `${process.env.WEATHER_URL}/weather`,
    params: {
      id,
      units: "metric",
      appid: "{WEATHER_APPID}"
    }
  });

  return data;
};
