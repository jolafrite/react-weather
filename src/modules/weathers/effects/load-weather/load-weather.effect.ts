import { Dispatch } from 'redux';
import { WeatherViewPageActions, WeatherApiActions } from '../../actions';
import { loadWeatherFetch } from '../../fetch';
import { loadWeatherFetchResponseTransformer } from '../../transformers';

export const viewPageLoadWeatherEffect = (id: number) => async (dispatch: Dispatch) => {
  dispatch(WeatherViewPageActions.loadOne());

  return await loadWeatherEffect(
    id,
    WeatherViewPageActions.loadOneSuccess,
    WeatherViewPageActions.loadOneError,
    dispatch
  );
}

export const loadWeatherEffect = async (id: number, onSuccess: any, onError: any, dispatch: Dispatch) => {
  try {
    const response = await loadWeatherFetch(id);
    const transformedResponse = loadWeatherFetchResponseTransformer(response);

    dispatch(WeatherApiActions.loadOneSuccess(transformedResponse));
    dispatch(onSuccess(transformedResponse));
  } catch(e) {
    dispatch(onError(e));
  }
};
