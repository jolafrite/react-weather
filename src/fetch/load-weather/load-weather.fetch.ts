import Axios from "axios";
import { IWeather } from "../../modules/weathers/models";

export const loadWeatherFetch = async (id: number): Promise<IWeather> => {
  const { data } = await Axios.request({
    method: "get",
    url: `${process.env.WEATHER_URL}/forecast`,
    params: {
      id,
      units: "metric",
      appid: "{WEATHER_APPID}"
    }
  });

  return data;
};
