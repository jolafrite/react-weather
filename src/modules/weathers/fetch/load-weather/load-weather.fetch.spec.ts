import Axios from "axios";
import { cleanup } from "@testing-library/react";
import { loadWeatherFetch } from "./load-weather.fetch";

jest.mock("axios");

describe('fetch: loadWeatherFetch', () => {

  afterEach(cleanup)

  describe('call with a query', () => {
    test('should call the api and return the response', async () => {
      const axiosRequest = (Axios.request as jest.Mock<any>);

      axiosRequest.mockResolvedValueOnce({
        data: 'the response'
      });

      const response = await loadWeatherFetch(111);

      expect(
        axiosRequest.mock.calls
      ).toMatchSnapshot();

      expect(
        response
      ).toMatchSnapshot()
    });
  });
});
