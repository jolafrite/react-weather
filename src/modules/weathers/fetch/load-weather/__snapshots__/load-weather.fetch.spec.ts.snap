// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch: loadWeatherFetch call with a query should call the api and return the response 1`] = `
Array [
  Array [
    Object {
      "method": "get",
      "params": Object {
        "appid": "{WEATHER_APPID}",
        "id": 111,
        "units": "metric",
      },
      "url": "undefined/weather",
    },
  ],
]
`;

exports[`fetch: loadWeatherFetch call with a query should call the api and return the response 2`] = `"the response"`;
